### PR TITLE
[hdf5] Fix tools

### LINKED ISSUE #27681: [hdf5] Build error during install cgns
Package: hdf5[core,szip,tools,zlib]:x64-linux -> 1.12.2#2

**Host Environment**

- OS:  x64-linux
- Compiler: GNU 12.2.1
- vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
   vcpkg-scripts version: 68ad399d5 2022-11-04 (2 days ago)

**To Reproduce**

This error happened when I downloaded cgns library, cgns library depends on hdf5[tools].But installing hdf5 alone does not trigger this error.

`vcpkg install cgns`

**Failure logs**

```
Computing installation plan...
The following packages will be built and installed:
    cgns[core,hdf5,lfs,lfsselector]:x64-linux -> 4.3.0#1
  * hdf5[core,szip,tools,zlib]:x64-linux -> 1.12.2#2
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/user/.cache/vcpkg/archives in 10.75 us. Use --debug to see more details.
Installing 1/2 hdf5:x64-linux...
Building hdf5[core,szip,tools,zlib]:x64-linux...
-- Using cached HDFGroup-hdf5-hdf5-1_12_2.tar.gz.
-- Cleaning sources at /home/user/.vcpkg/buildtrees/hdf5/src/df5-1_12_2-c0e67ca243.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/user/.vcpkg/downloads/HDFGroup-hdf5-hdf5-1_12_2.tar.gz
-- Applying patch hdf5_config.patch
-- Applying patch szip.patch
-- Applying patch pkgconfig-requires.patch
-- Applying patch pkgconfig-link-order.patch
-- Using source at /home/user/.vcpkg/buildtrees/hdf5/src/df5-1_12_2-c0e67ca243.clean
-- Found external ninja('1.10.2').
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Fixing pkgconfig file: /home/user/.vcpkg/packages/hdf5_x64-linux/lib/pkgconfig/hdf5.pc
-- Fixing pkgconfig file: /home/user/.vcpkg/packages/hdf5_x64-linux/lib/pkgconfig/hdf5_hl.pc
-- Fixing pkgconfig file: /home/user/.vcpkg/packages/hdf5_x64-linux/debug/lib/pkgconfig/hdf5.pc
-- Fixing pkgconfig file: /home/user/.vcpkg/packages/hdf5_x64-linux/debug/lib/pkgconfig/hdf5_hl.pc
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "h5c++":

      "/home/user/.vcpkg/packages/hdf5_x64-linux/bin/h5c++" does not exist
Call Stack (most recent call first):
  ports/hdf5/portfile.cmake:138 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)
```

**Additional context**

This problem didn't happen with hdf5 1.12.2#1, the above error happened when I updated vcpkg and tried to update the package.

### PR DESCRIPTION
**Pull request by Adela0814**
Fixes https://github.com/microsoft/vcpkg/issues/27681

Error message:
```
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "h5c++":

      "/home/user/.vcpkg/packages/hdf5_x64-linux/bin/h5c++" does not exist
```
The `h5c++` and `h5hlc++` stuff needs to be behind the `cpp` feature.

### CODE REVIEW

File: ports/hdf5/portfile.cmake [hunk 1]
```diff
@@ -117,7 +117,7 @@ if("tools" IN_LIST FEATURES)
     endif()
 
     if(NOT VCPKG_TARGET_IS_WINDOWS)
-        list(APPEND HDF5_TOOLS h5cc h5hlcc h5c++ h5hlc++)
+        list(APPEND HDF5_TOOLS h5cc h5hlcc)
```
Comment (Neumann-A) [inline ports/hdf5/portfile.cmake:120]:
```suggestion
        list(APPEND HDF5_TOOLS h5cc h5hlcc)
        if("cpp" IN_LIST FEATURES)
                list(APPEND HDF5_TOOLS h5c++ h5hlc++)
        endif()
```

File: versions/baseline.json [hunk 2]
```diff
@@ -6832,6 +6832,10 @@
       "baseline": "1.3.1",
       "port-version": 0
     },
+    "shader-slang": {
+      "baseline": "0.23.13",
+      "port-version": 0
+    },
```
Comment (Neumann-A) [inline versions/baseline.json:6838]: unrelated diff ?

Comment (Adela0814) [reply versions/baseline.json]: Merged the latest master to restore this.

**Review by FrankXie05 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!
